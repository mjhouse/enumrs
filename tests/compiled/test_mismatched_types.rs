use enumrs::Tagged;

#[derive(Tagged)]
pub enum TestEnum {

    #[tag(tagname)]
    Variant1 = 1,

    // Tags on different variants that have
    // different types will fail to compile
    // because we can't match and return the
    // values from the same function.
    #[tag(tagname, 3.5)]
    Variant2
}

pub fn main() {}