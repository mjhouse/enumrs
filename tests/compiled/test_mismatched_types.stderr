error: Mismatched expression value: expected 'bool', found 'f64'
  --> tests/compiled/test_mismatched_types.rs:13:5
   |
13 |     #[tag(tagname, 3.5)]
   |     ^
